pkg/devfile/adapters/kubernetes/component/adapter_test.go

```diff
@@ -555,6 +569,148 @@ func TestAdapterDelete(t *testing.T) {
 	}
 }
 
+func TestCreateDockerConfigSecretBytes(t *testing.T) {
+
+	testConfigJsonString := "{ \"auths\" : { \"https://index.docker.io/v1/\": { \"auth\": \"test-auth-token\", \"email\": \"test-email\"} },\"HttpHeaders\": {	\"User-Agent\": \"Docker-Client/19.03.8 (darwin)\"},\"experimental\": \"disabled\"}"
+	testFs := NewFilesystem()
+	testFilename := "test-config-json"
+	testConfigJsonBytes := []byte(testConfigJsonString)
+
+	defer testFs.Remove(testFilename)
+
+	afero.WriteFile(testFs, testFilename, []byte(testConfigJsonString), 0777)
+
+	fkclient, _ := kclient.FakeNew()
+	testAdapter := Adapter{
+		Client: *fkclient,
+	}
+
+	want := testConfigJsonBytes
+	got, err := testAdapter.createDockerConfigSecretBytes(testFilename)
+	if err != nil {
+		t.Error(err)
+		t.Errorf("failed to retrieve dockerconfig secret bytes")
+	}
+
+	if !reflect.DeepEqual(want, got) {
+		t.Errorf("secret bytes don't match")
+	}
+}
+
+func TestCreateDockerConfigSecret(t *testing.T) {
+	testConfigJsonString := "{ \"auths\" : { \"https://index.docker.io/v1/\": { \"auth\": \"test-auth-token\", \"email\": \"test-email\"} },\"HttpHeaders\": {	\"User-Agent\": \"Docker-Client/19.03.8 (darwin)\"},\"experimental\": \"disabled\"}"
```

Comment:
Create a test-data/config.json and pass the file path (relative path "test-data/config.son") to the Parameters which is input to createDockerConfigSecret()

Reply:
Done